src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.cs

```diff
@@ -12,14 +17,404 @@ internal partial struct SslConnectionInfo
         private void MapCipherSuite(TlsCipherSuite cipherSuite)
         {
             TlsCipherSuite = cipherSuite;
-
-            TlsCipherSuiteData data = TlsCipherSuiteData.GetCipherSuiteData(cipherSuite);
-            KeyExchangeAlg = (int)data.KeyExchangeAlgorithm;
             KeyExchKeySize = 0;
-            DataCipherAlg = (int)data.CipherAlgorithm;
-            DataKeySize = data.CipherAlgorithmStrength;
-            DataHashAlg = (int)data.MACAlgorithm;
-            DataHashKeySize = data.MACAlgorithmStrength;
+
+            ushort data = GetPackedData(cipherSuite);
+            Debug.Assert(data != 0, $"No mapping found for cipherSuite {cipherSuite}");
+
+            KeyExchangeAlg = (int)s_exchangeAlgorithmTypes[(data >> (16 - (4 * 1)) & 0xF)];
+            DataCipherAlg = (int)s_cipherEnumValues[(data >> (16 - (4 * 2)) & 0xF)];
+            DataKeySize = (int)s_cipherStrengthEnumValues[(data >> (16 - (4 * 3)) & 0xF)];
+            DataHashAlg = (int)s_hashEnumValues[(data >> (16 - (4 * 4)) & 0xF)];
+            DataHashKeySize = GetHashSize((HashAlgorithmType)DataHashAlg);
+
+            static int GetHashSize(HashAlgorithmType hash)
+            {
+                switch (hash)
+                {
+                    case HashAlgorithmType.None:
+                        return 0;
+                    case HashAlgorithmType.Md5:
+                        return 128;
+                    case HashAlgorithmType.Sha1:
+                        return 160;
+                    case HashAlgorithmType.Sha256:
+                        return 256;
+                    case HashAlgorithmType.Sha384:
+                        return 384;
+                    case HashAlgorithmType.Sha512:
+                        return 512;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(hash));
+                }
+            }
+        }
+
+        private static ReadOnlySpan<int> s_exchangeAlgorithmTypes =>
```

Comment:
This is a property, not a field, so it should not have `s_` prefix. We do not need a static property for this at all. The map can be defined inline in the method as a local variable like this: https://github.com/dotnet/runtime/pull/84823/files#diff-7b7cfb041361d0910a9256b87bc237341efb6182e04e3cf34ae860fe1b27de23R342

Reply:
Changed to local variable.